From cf4ed917e2d7d6b3683ba1aa4ea4662538b69516 Mon Sep 17 00:00:00 2001
From: Nito Martinez <Nito@Qindel.ES>
Date: Fri, 15 May 2015 22:40:05 +0200
Subject: [PATCH] This patch is some code fixes to debug some debuging macro
 usage in the NX code.

Particularly the following macros have been tested -DTEST -DDEBUG -DDUMP -DFLUSH -DTOKEN -DSPLIT -DPING -DMIXED -DMATCH -DTIME
---
 nxcomp/Agent.h       | 2 +-
 nxcomp/ClearArea.cpp | 2 +-
 nxcomp/Proxy.cpp     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/nxcomp/Agent.h b/nxcomp/Agent.h
index ded344d..2c5badf 100644
--- a/nxcomp/Agent.h
+++ b/nxcomp/Agent.h
@@ -217,7 +217,7 @@ class Agent
 
     #if defined(TEST) || defined(INFO)
     *logofs << "Agent: proxyCanRead() is "
-            << ((int) FD_ISSET(proxy -> getFd(), &readWorkSet)
+            << ((int) FD_ISSET(proxy -> getFd(), &readWorkSet))
             << ".\n" << logofs_flush;
     #endif
 
diff --git a/nxcomp/ClearArea.cpp b/nxcomp/ClearArea.cpp
index 223a3b3..83b7281 100644
--- a/nxcomp/ClearArea.cpp
+++ b/nxcomp/ClearArea.cpp
@@ -92,7 +92,7 @@ void ClearAreaStore::dumpIdentity(const Message *message) const
 
   ClearAreaMessage *clearArea = (ClearAreaMessage *) message;
 
-  *logofs << name() << ": Identity exposures " << clearArea -> (unsigned int) exposures 
+  *logofs << name() << ": Identity exposures " << (unsigned int) clearArea -> exposures 
           << ", window " << clearArea -> window  << ", x " << clearArea -> x 
           << ", y " << clearArea -> y << ", width  " << clearArea -> width 
           << ", height " << clearArea -> height << ", size " << clearArea -> size_ 
diff --git a/nxcomp/Proxy.cpp b/nxcomp/Proxy.cpp
index 9b38661..4eb99cf 100644
--- a/nxcomp/Proxy.cpp
+++ b/nxcomp/Proxy.cpp
@@ -5176,7 +5176,7 @@ char *Proxy::handleSaveAllStores(const char *savePath) const
 
   *(cacheDumpName + DEFAULT_STRING_LENGTH - 1) = '\0';
 
-  mode_t fileMode = umask(0077);
+  fileMode = umask(0077);
 
   cacheDump = new ofstream(cacheDumpName, ios::out);
 
